python/cudf/cudf/core/reshape.py

```diff
@@ -63,6 +91,9 @@ def concat(objs, axis=0, ignore_index=False, sort=None):
         assert typs.issubset(allowed_typs)
         df = DataFrame()
 
+        if not ignore_index:
+            objs = align_objs(objs)
```

Comment:
In hindsight, I believe I misguided you on how to handle the `ignore_index` parameter. According to pandas docs: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.concat.html, even if `ignore_index=True` the indexes are still respected for the join, but the output index will be 0 --> N. So we always have to do this alignment.

Reply:
Below on line 108 though we only need to set `df.index = o.index` if  `ignore_index` is False.

Reply:
Ok. I suspected that it always needed to be aligned but I wasn't sure as I've also been told we don't have to do what pandas does. Do I need to sort the index too? Currently I'm sorting the index in the test. I wanted to save the operation if it wasn't required, but it seems to me that sorting the output on L117 would be desirable.

Reply:
Oh good point... Since we're using a join for alignment we may get back a semi-arbitrary order. What does Pandas do in this situation?

Reply:
pandas sorts.

Reply:
I'm guessing that's more of a side effect of their join algorithm, but lets sort for now. We can revisit optimizing in the future.